localstack/services/awslambda/lambda_executors.py

```diff
@@ -463,7 +465,17 @@ def _run(func_arn=None):
             FuncThread(do_execute).start()
             return InvocationResult(None, log_output="Lambda executed asynchronously.")
 
-        return do_execute()
+        lock = LAMBDA_ASYNC_LOCKS.locks[lock_discriminator]
+        acquired = False
+        try:
+            acquired = lock.acquire(blocking=False)
```

Comment:
note: a sync call needs to fail immediately when it can't acquire the lock here instead of waiting for one to be freed.